src/http.rs

```diff
@@ -80,13 +109,19 @@ impl Client {
     }
 
     pub async fn fetch_stream(&self, url: &str) -> Result<Download> {
```

Comment:
This function should get a new `offset: Option<u64>` argument that allows restarting the download at an offset. You also need to extend `send_get` to support this. If `offset.is_none()` the Range header should not be added.